Makefile

```
@@ -148,51 +134,20 @@ test: core
 
 ########################## MAKE/INSTALL RULES ##########################
 #
-all: dependencies core
+all: core
 
 core:
-	@command -v $(firstword $(MSBUILD)) >/dev/null || (echo "OpenRA requires the '$(MSBUILD)' tool provided by Mono >= 5.4."; exit 1)
-ifeq ($(WIN32), $(filter $(WIN32),true yes y on 1))
-	@$(MSBUILD) -t:build -p:Configuration="Release-x86"
-else
-	@$(MSBUILD) -t:build -p:Configuration=Release
+	@command -v $(firstword $(MSBUILD)) >/dev/null || (echo "OpenRA requires the '$(MSBUILD)' tool provided by Mono >= 5.18."; exit 1)
+	@$(MSBUILD) -t:Build -restore -p:Configuration=Release -p:TargetPlatform=$(TARGETPLATFORM)
```

On a second thought I think this is what causes my fails. I wanted to submit a patch then noticed my base was outdated. If you don't restore and build at the same time, msbuild can't error when switching .NET versions:

```suggestion
	@$(MSBUILD) -t:restore
	@$(MSBUILD) -t:Build -p:Configuration=Release -p:TargetPlatform=$(TARGETPLATFORM)
```

You should also separate building and dependency fetching for Linux distributions who don't want a build script to connect to the internet and download binaries on it's own, which is seen as a security threat.

---

> 
> 
> On a second thought I think this is what causes my fails. I wanted to submit a patch then noticed my base was outdated. If you don't restore and build at the same time, msbuild can't error when switching .NET versions:
> 

you mean that msbuild will report an error when using your suggested change, that it didnt do with the original code?

> You should also separate building and dependency fetching for Linux distributions who don't want a build script to connect to the internet and download binaries on it's own, which is seen as a security threat.

This was was discussed earlier, and the conclusion then was to do it one task.

---

If it won't connect to the internet when the dependencies are already there, then that is fine. However, having restore and build in one command seems to have caused the failures where it didn't restore, because the build failed and the build failed because it didn't restore properly. As make clean also failed there was no way to solve this other than `dotnet restore` manually for me.